src/common/k8s-api/api-kube.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { isClusterPageContext } from "../utils";
+import { KubeJsonApi } from "./kube-json-api";
+import { apiKubePrefix, isDevelopment } from "../vars";
+
+export const apiKube = isClusterPageContext()
+  ? new KubeJsonApi({
+    serverAddress: `http://127.0.0.1:${window.location.port}`,
+    apiBase: apiKubePrefix,
+    debug: isDevelopment,
+  }, {
+    headers: {
+      "Host": window.location.host,
+    },
+  })
+  : null;
```

Comment:
No that is not part of the solution. I can change it back. There shouldn't be except that the error would now be `Cannot read property X of null` instead of `Cannot read property X of undefined`

Reply:
Could break an extension using `if ( apiKube !== undefined )`? Probably not a big deal (they should use `if ( apiKube )`) but why change it unless `null` is somehow better?

Reply:
It certainly isn't better, though some people I think argue that `null` should represent a deliberate case of not present but I will revert it.